peer_test.go

```diff
@@ -189,6 +189,48 @@ func TestPeerRemoveClosedConnection(t *testing.T) {
 	})
 }
 
+func TestPeerGetConnectionWithNoActiveConnections(t *testing.T) {
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+
+	WithVerifiedServer(t, nil, func(ch *Channel, hostPort string) {
+		client := testutils.NewClient(t, nil)
+		defer client.Close()
+
+		var (
+			wg          sync.WaitGroup
+			lock        sync.Mutex
+			conn        *Connection
+			numRequests = 10
+			p           = client.Peers().Add(hostPort)
+		)
+
+		for i := 0; i < numRequests; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				c, err := p.GetConnection(ctx)
+				require.NoError(t, err, "GetConnection failed")
+
+				lock.Lock()
+				defer lock.Unlock()
+
+				if conn == nil {
```

Comment:
This is a data race since multiple goroutines will be checking `conn`, suggest instead using a slice,
```go
conns := make([]*Connection, concurrency)
[....]
go func(i int) {
  [...]
  conns[i] = c
}
```

and then after the `wg.Wait`, verify that they're all the same